test/not_null.t.cpp

```diff
@@ -274,6 +326,34 @@ CASE( "not_null<>: Allows to construct from a non-null underlying pointer (share
     EXPECT( p == pi );
 }
 
+CASE( "not_null<>: Returns underlying pointer with get() (shared_ptr)" )
+{
+    shared_ptr< int > pi = make_shared< int >(12);
+    not_null< shared_ptr< int > > p( pi );
+
+#if gsl_CONFIG( NOT_NULL_TRANSPARENT_GET )
+    int* pg = p.get();
+    EXPECT( pg == pi.get() );
+#else
+    shared_ptr< int > const & pg = p.get();
+    EXPECT( pg == pi );
+#endif
+}
+
+CASE( "not_null<>: Allows to move from a not_null pointer (shared_ptr)" )
+{
+#if gsl_HAVE( FUNCTION_REF_QUALIFIER )
+    shared_ptr< int > pi = make_shared< int >(12);
+    int* raw(pi.get());
+
+    not_null< shared_ptr< int > > p ( std::move(pi) ); // There...
+    pi = std::move(p); // ...and back again.
+
+    EXPECT_THROWS( *p );
+    EXPECT( pi.get() == raw );
+#endif
+}
```

Comment:
Why is this test only supported when FUNCTION_REF_QUALIFIER is supported? This should work on gcc 4.7.3 too!

Also, the test case's name is misleading: it does not mention that we're moving to an underlying pointer.

Reply:
I fail to see how this could work without `FUNCTION_REF_QUALIFIER`. The conversion operator plainly fails for `not_null<unique_ptr<T>>` because it cannot call `move()`.